learnify/mobile/lib/features/search/view/components/search_bar_widget.dart

```diff
@@ -0,0 +1,64 @@
+part of '../search_screen.dart';
+
+class SearchBarWidget extends StatefulWidget {
```

Comment:
from what I see in your code, you don't make use of Stateful properties such as SetState. Does this widget need to be Stateful?